### DISCUSSION TITLE
Will supabase be here for a long time? [Asking for a friend]

### DISCUSSION THREAD

**Discussion by rotimi-best** [supabase/supabase#2563] [Questions] [answered]
I am building a side project, that I plan on scaling cause I am beginning to have potential customers.

I am so pumped about the features that supabase offers, mostly because I can manage my database from a UI without any downloads or connection. My fear is that if I start building on top of supabase and the project ends up dieing, it will be really painful to migrate all the code integration in the project. This is my first Saas which justifies my worry and shows I don't have that experience on what it really means to scale (which is why I am asking for a friend).

Is supabase here to stay? Is there a good chance, I can build my whole project on top of supabase and it will scale with me?

Thank you to the team working on this product, I think it is worth the price.

**Comment by koaning** [comment, marked as answer]
I'm not an expert, but Postgres (which is what supabase is built on) can handle quite a bit. Reddit uses it. IMDB uses it. So unless you've got an exceptional amount of data you'd like to process upfront I wouldn't be that worried. 

In terms of being "here for a long time", the project is hosted here as open-source too. Should the company ever go down, the source code (which is Apache License 2.0) can be hosted by your own team as well. This might certainly not be ideal, but it's at least possible.

**Reply by nischalmudennavar** [reply]
it will take some time for devs to actually use it for their production builds. As a MERN stack developer .....I usually come across a situation where I have to rely on s3 bucket or firebase for media storage. and setting up that ( UI wise ) is hard..... whereas Supabase has an easy way of doing those things ! So yeah I guess it can be up for the long haul!

**Reply by rotimi-best** [reply]
@koaning @nischalmudennavar You both mention great points. Thanks for taking the time to answer. I really appreciate it.
